rclcpp/include/rclcpp/rate.hpp

```diff
@@ -33,46 +33,43 @@ class RateBase
 
   virtual ~RateBase() {}
   virtual bool sleep() = 0;
-  virtual bool is_steady() const = 0;
+  virtual rcl_clock_type_t get_type() const = 0;
   virtual void reset() = 0;
 };
 
-using std::chrono::duration;
-using std::chrono::duration_cast;
-using std::chrono::nanoseconds;
-
-template<class Clock = std::chrono::high_resolution_clock>
-class GenericRate : public RateBase
+class Rate : public RateBase
 {
 public:
-  RCLCPP_SMART_PTR_DEFINITIONS(GenericRate)
+  RCLCPP_SMART_PTR_DEFINITIONS(Rate)
 
-  explicit GenericRate(double rate)
```

Comment:
I think we need to deprecate this for this moment with following message.


``[[deprecated("use rclcpp::Rate class instead of GenericRate")]]``

Reply:
Done in https://github.com/ros2/rclcpp/pull/2123/commits/c25fe6bdad6ee686fa411a89cd7ae140e0f3ac40 and https://github.com/ros2/rclcpp/pull/2123/commits/7323e9730a8f3e288177bb30b74777f34e33dfee. `GenericRate`'s API remained to be the same in order to make the compiler to inform developer only about class name change. Implementation was cleaned-out.